hal/platform/atmelsam/irq/sercom_stuff.h

```diff
@@ -33,9 +33,17 @@ typedef struct {
     uint8_t* read_buffer;
     uint8_t buf_size;
     uint8_t buf_cnt;
+    uint8_t rx_status;
```

Comment:
It might be better to make this one variable called busstatus and use masks. Now when ISR is started it has to load 120 bytes of data from memory to use. With the possibility of making the ISR performance sluggish. Another thing is that creating 8-bits variables for 1-bit purpose, leaves 7-bits unused. So it might be worth it from space saving standpoint as well